File: src/main/java/MorontaSharp.java
import java.util.InputMismatchException;
import java.util.Scanner;

public class MorontaSharp {
    public static void main(String[] args) {
        boolean exit = false;
        boolean valido = true;

        do{
            int n;
            int x = 0;
            String sentencia;
            long startTime = System.nanoTime();

            Scanner scanner = new Scanner(System.in);
            System.out.println("Inserte la cantidad de sentencias ( 1 - 150 ): ");
            try {
                n = scanner.nextInt();
                if(n >= 1 && n <= 150){
                    do{

                        System.out.println("Inserte la sentencia " + n);
                        sentencia = scanner.next();

                        long stopTime = System.nanoTime();
                        long elapsedTime = (stopTime - startTime)/1_000_000_000;
                        System.out.println("Time : " + elapsedTime + " seconds");

                        if(sentencia.length() == 3){
                            switch (sentencia){
                                case "X++":
                                    x++;
                                    n--;
                                    break;
                                case "++X":
                                    ++x;
                                    n--;
                                    break;
                                case "X--":
                                    x--;
                                    n--;
                                    break;
                                case "--X":
                                    --x;
                                    n--;
                                    break;
                                default:
                                    System.out.println("Sentencia no válida");
                                    break;
                            }
                        }else{
                            System.out.println("Sentencia no válida");
                        }
                        if(n == 0){
                            valido = true;
                            exit = true;
                        }

                    }while (!exit);
                    System.out.println("El resultado es: " + x);
                }else{
                    System.out.println("Por favor, inserte una cantidad de sentencias válida");
                }
            }catch (InputMismatchException ime){
                System.out.println("Por favor, introduzca un número");
                valido =false;
            }

        }while (!valido);
    }
}

File: Go/morontaSharp.go
package main

import "fmt"
import "bufio"
import "strconv"
import "os"
import "time"

func main() {

  start := time.Now()

  scanner := bufio.NewScanner(os.Stdin)
  exit := false
  x := 0
  fmt.Println("Inserte la cantidad de sentencias ( 1 - 150 ):")
  scanner.Scan()
  
  if n,error := strconv.Atoi(scanner.Text()); error == nil{
    if n >= 1 && n <= 150{
      for{
        fmt.Println("Inserte la sentencia " + strconv.Itoa(n))
        scanner.Scan()
        sentencia := scanner.Text()
          if len(sentencia) == 3{
            switch sentencia{
              case "X++":
                x++
                n--
              case "++X":
                x++
                n--
              case "X--":
                x--
                n--
              case "--X":
                x--
                n--
              default:
                fmt.Println("Sentencia no válida")
            }

          }else{
            fmt.Println("Sentencia no válida")
          }
        if n == 0{
          fmt.Println("El resultado es: " + strconv.Itoa(x))
          exit = true
        }

        if(exit){
          break
        }
      }
    }else{
      fmt.Println("Por favor, inserte una cantidad de sentencias válida")
    }
  }else{
    fmt.Println("Por favor, introduzca un número")
  }
elapsed:=time.Since(start)
fmt.Printf("Time : %s \n",elapsed)
}
